mllib-dal/src/main/scala/com/intel/oap/mllib/OneDAL.scala

```diff
@@ -668,6 +668,29 @@ object OneDAL {
     }
   }
 
+  def coalesceToHomogenTables(data: RDD[Vector], executorNum: Int,
+                                device: Common.ComputeDevice): RDD[Long] = {
+    // Coalesce partitions belonging to the same executor
+    val coalescedRdd = data
+      .coalesce(executorNum,
+        partitionCoalescer = Some(new ExecutorInProcessCoalescePartitioner()))
+      .setName("Repartitioned for conversion")
+      .cache()
+
+    // Unpersist instances RDD
+    if (data.getStorageLevel != StorageLevel.NONE) {
```

Comment:
There is a problem here. above coalescedRdd cache() is not an action. It will not be executed immediately if you don't insert an action such as count(). I think you need to unpersist parent RDD after child RDD is cached.